web/nginx/nginx.conf

```diff
@@ -74,8 +75,14 @@ http {
             proxy_no_cache $http_pragma     $http_authorization;
             proxy_cache_bypass $cookie_nocache $arg_nocache $arg_comment;
             proxy_cache_bypass $http_pragma $http_authorization;
-            if ($http_x_forwarded_proto != 'https') {
-                rewrite ^ https://$server_name$request_uri?;
+            if ($host = 'api.masschallenge.org'){
+             set $local_and_insecure  "prod"; 
+            }
+            if ($http_x_forwarded_proto != 'https'){
+             set $local_and_insecure  "${local_and_insecure}insecure"; 
+            }
+            if ($local_and_insecure = 'prodinsecure'){
+                    rewrite ^ https://$host$request_uri?;
```

Comment:
Lines 78-85 redirect from http to https, but only if the host is api.masschallenge.org -- is that right?  Might be worth a comment above the block (especially if I'm wrong!)

Reply:
good call (and you are correct). comment added.